pgx/src/spi.rs

```diff
@@ -378,8 +378,176 @@ impl SpiClient {
             current: -1,
         }
     }
+
+    /// sets up a cursor that will execute the specified query.
+    /// Rows may be then fetched using [`SpiCursor::fetch`]
+    ///
+    /// See [`SpiCursor`] docs for usage details.
+    pub fn open_cursor(
+        &mut self,
+        query: &str,
+        args: Option<Vec<(PgOid, Option<pg_sys::Datum>)>>,
+    ) -> SpiCursor {
+        let src = std::ffi::CString::new(query).expect("query contained a null byte");
+        let args = args.unwrap_or(vec![]);
+
+        let nargs = args.len();
+        let mut argtypes = vec![];
+        let mut datums = vec![];
+        let mut nulls = vec![];
+
+        for (argtype, datum) in args {
+            argtypes.push(argtype.value());
+
+            match datum {
+                Some(datum) => {
+                    // ' ' here means that the datum is not null
+                    datums.push(datum);
+                    nulls.push(' ' as std::os::raw::c_char);
+                }
+
+                None => {
+                    // 'n' here means that the datum is null
+                    datums.push(0);
+                    nulls.push('n' as std::os::raw::c_char);
+                }
+            }
+        }
+
+        let ptr = unsafe {
+            pg_sys::SPI_cursor_open_with_args(
+                std::ptr::null_mut(), // let postgres assign a name
+                src.as_ptr(),
+                nargs as i32,
+                argtypes.as_mut_ptr(),
+                datums.as_mut_ptr(),
+                nulls.as_ptr(),
+                false,
+                0,
+            )
+        };
+        SpiCursor { ptr }
+    }
+
+    /// find a cursor in transaction by name.
+    /// Name should usually be retrieved via [`SpiCursor::into_name`]
+    ///
+    /// See [`SpiCursor`] docs for usage details.
+    pub fn find_cursor(
+        &mut self,
+        name: String,
+    ) -> SpiCursor {
+        use pgx_pg_sys::AsPgCStr;
+
+        let ptr = unsafe {
+            pg_sys::SPI_cursor_find(name.as_pg_cstr())
+        };
+        if ptr.is_null() {
+            error!("cursor named \"{}\" not found", name);
+        }
+        SpiCursor { ptr }
+    }
+}
+
+/// Represents a Postgres cursor (internally, a portal), allowing to retrieve result rows a few
```

Comment:
`cargo doc` will give significant prominent in module level docs to the first line which should be a short sentence describing the role of the type. It should not have a period at the end.

```suggestion
/// An SPI cursor from a query
///
/// Represents a Postgres cursor (internally, a portal), allowing to retrieve result rows a few
```